### PR TITLE
Update adapter to use the 3.5 engine

### LINKED ISSUE #236: Update adapter to use Version 3.5 of the Engine

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #236

### CODE REVIEW

**Comment by CharliePoole** [comment]
@rprouse @OsirisTerje Because this PR changes the cake version and bootstrap script I'm basing my other 3.5 changes off it. If it looks good, I'd appreciate if one of you would merge it as soon as you can.

File: src/NUnitTestAdapterTests/TestDiscoveryTests.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public void LoadMockassembly()
             Assert.That(NUnit.Tests.Assemblies.MockAssembly.Tests, Is.EqualTo(31),
                 "The reference to mock-assembly.dll appears to be the wrong version");
 
+            TestCases = new List<TestCase>();
```
Comment (rprouse) [inline src/NUnitTestAdapterTests/TestDiscoveryTests.cs]: This isn't being used as far as I can see. I assume it is getting used in a subsequent PR?

File: src/NUnitTestAdapter/NUnit3TestDiscoverer.cs [hunk 2]
```diff
@@ -53,15 +53,14 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
 
                 try
                 {
-                    XmlNode loadResult = runner.Load();
+                    XmlNode topNode = runner.Explore(TestFilter.Empty);
```
Comment (rprouse) [inline src/NUnitTestAdapter/NUnit3TestDiscoverer.cs]: Why the change? Just curious...

File: src/NUnitTestAdapterTests/TestDiscoveryTests.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitTestAdapterTests/TestDiscoveryTests.cs]: Actually, it is. The test class implements ITestCaseDiscoverySink and receives events as each test is found. The tests are saved in the list for use in the test methods.
